.github/workflows/pr-auto-format.yml

```diff
@@ -0,0 +1,205 @@
+name: PR Auto-format
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
+      frontend: ${{ steps.filter.outputs.frontend }}
+      models: ${{ steps.filter.outputs.models }}
+      templates: ${{ steps.filter.outputs.templates }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            backend:
+              - 'app/backend/**'
+            frontend:
+              - 'app/web/**'
+            models:
+              - 'app/backend/src/couchers/models/**'
+            templates:
+              - 'app/backend/templates/v2/**'
+              - '!app/backend/templates/v2/generated_html/**'
+
+  format-backend:
+    needs: detect-changes
+    if: needs.detect-changes.outputs.backend == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is this token distinct from `github.token`? It needs to be to retrigger CI after the push.